File: February challenges/Python/2966 Divide Array Into Arrays With Max Difference.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        n = len(nums)
        nums.sort()
        res = []
        i = 0

        while i < n:
            new_nums = nums[i:i+3]    
            if abs(new_nums[0] - new_nums[-1]) <= k:
                res.append(new_nums)
            else:
                res.clear()
                break
            i += 3
    
        return res

File: February challenges/Java/2966 Divide Array Into Arrays With Max Difference.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        int n = nums.length;
        int i = 0;
        Arrays.sort(nums);
        ArrayList<int[]> res = new ArrayList<>();
        while(i < n){
            int[] subarray = {nums[i], nums[i+1], nums[i+2]};

            if(Math.abs(subarray[0] - subarray[2]) <= k){
                res.add(subarray);
            }
            else{
                res.clear();
                break;
            }
            i += 3;
        }
        return res.toArray(new int[res.size()][]);
        
    }
}
